src/controller/src/lib.rs

```diff
@@ -457,6 +460,7 @@ where
     ) -> Result<(), anyhow::Error> {
         if let Some(compute_state) = self.compute.remove(&instance) {
             compute_state.drop();
+            self.compute_stash.cleanup_instance(instance).await?;
```

Comment:
What happens if we crash right after this line? Does that leak compute_orchestrator resources?